luavm/src/lib/instructions/loads.rs

```diff
@@ -1,11 +1,12 @@
 use lua_value::*;
 use luacompiler::bytecode::instructions::{first_arg, second_arg};
+use std::mem;
 use Vm;
 
 pub fn mov(vm: &mut Vm, instr: u32) {
     let i = first_arg(instr) as usize;
     let j = second_arg(instr) as usize;
-    vm.registers.swap(i, j);
+    vm.registers[i] = mem::replace(&mut vm.registers[j], Box::new(LuaNil {}));
```

Comment:
Why does a move `Nil` the old register?

Reply:
Ah, now that I think about it, it should make a copy, and not move the value.